src/main/java/org/jruby/ext/openssl/ASN1.java

```diff
@@ -1072,24 +1072,23 @@ else if ( obj instanceof DERBMPString ) {
             return ASN1.getClass("ObjectId").newInstance(context, runtime.newString(objId), Block.NULL_BLOCK);
         }
 
-        if ( obj instanceof ASN1ApplicationSpecific ) { // TODO this will likely break in BC version > 1.71
-            final ASN1ApplicationSpecific appSpecific = (ASN1ApplicationSpecific) obj;
-            IRubyObject tag = runtime.newFixnum( appSpecific.getApplicationTag() );
-            IRubyObject tag_class = runtime.newSymbol("APPLICATION");
-            final ASN1Sequence sequence = (ASN1Sequence) appSpecific.getObject(SEQUENCE);
-            @SuppressWarnings("unchecked")
-            final RubyArray valArr = decodeObjects(context, ASN1, sequence.getObjects());
-            return ASN1.getClass("ASN1Data").newInstance(context, new IRubyObject[] { valArr, tag, tag_class }, Block.NULL_BLOCK);
-        }
-
-        if ( obj instanceof ASN1TaggedObject ) {
-            final ASN1TaggedObject taggedObj = (ASN1TaggedObject) obj;
-            IRubyObject val = decodeObject(context, ASN1, taggedObj.getBaseObject());
-            IRubyObject tag = runtime.newFixnum( taggedObj.getTagNo() );
-            IRubyObject tag_class = runtime.newSymbol("CONTEXT_SPECIFIC");
-            final RubyArray valArr = runtime.newArray(val);
-            return ASN1.getClass("ASN1Data").newInstance(context, new IRubyObject[] { valArr, tag, tag_class }, Block.NULL_BLOCK);
-        }
+        if (obj instanceof ASN1TaggedObject) {
```

Comment:
formatting does show-up a bit weird for whatever reason, thus:
```suggestion
        if (obj instanceof ASN1TaggedObject) {
	        final ASN1TaggedObject taggedObj = (ASN1TaggedObject) obj;
            if (taggedObj.getTagClass() == BERTags.APPLICATION) {
                IRubyObject tag = runtime.newFixnum( taggedObj.getTagNo() );
                IRubyObject tag_class = runtime.newSymbol("APPLICATION");
                final ASN1Sequence sequence = (ASN1Sequence) taggedObj.getBaseUniversal(false, SEQUENCE);
                @SuppressWarnings("unchecked")
                final RubyArray valArr = decodeObjects(context, ASN1, sequence.getObjects());
                return ASN1.getClass("ASN1Data").newInstance(context, new IRubyObject[] { valArr, tag, tag_class }, Block.NULL_BLOCK);
	        } else {
                IRubyObject val = decodeObject(context, ASN1, taggedObj.getBaseObject());
                IRubyObject tag = runtime.newFixnum( taggedObj.getTagNo() );
                IRubyObject tag_class = runtime.newSymbol("CONTEXT_SPECIFIC");
                final RubyArray valArr = runtime.newArray(val);
                return ASN1.getClass("ASN1Data").newInstance(context, new IRubyObject[] { valArr, tag, tag_class }, Block.NULL_BLOCK);
            }
        }
```

Reply:
When I merged this it then duplicated the conditional, so I had to rework it. I think the previous alignment issue was a tabs vs spaces thing. Hopefully it's resolved now.